File: Swift/ExampleTests/common/ResponseTyped.swift
public struct ResponseTyped: Equatable, CustomStringConvertible {
    public let result: ResultTyped

    public init(result: ResultTyped) {
        self.result = result
    }

    public init(from s: ResponseString) {
        self.result = ResultTyped(from: s.result)
    }

    public func toStringStruct() -> ResponseString {
        return ResponseString(
            result: result.toStringStruct()
        )
    }

    public static func toStringList(_ list: [ResponseTyped]) -> [ResponseString] {
        return list.map { $0.toStringStruct() }
    }

    public static func fromStringList(_ list: [ResponseString]) -> [ResponseTyped] {
        return list.map { ResponseTyped(from: $0) }
    }

    public func toJSONValue() -> [String: Any] {
        return [
            "result": result.toJSONValue(),
        ]
    }

    public func toJSON() throws -> String {
        return try Json.write(toJSONValue())
    }

    public init(fromJSONValue m: [String: Any]) throws {
        self.result = try ResultTyped(fromJSONValue: Json.asObject(Json.require(m, "result"), "result"))
    }

    public init(fromJSON text: String) throws {
        try self.init(fromJSONValue: Json.parseObject(text))
    }

    public static func toJSONList(_ list: [ResponseTyped]) throws -> String {
        return try Json.write(list.map { $0.toJSONValue() })
    }

    public static func fromJSONList(_ text: String) throws -> [ResponseTyped] {
        return try Json.parseArray(text).map {
            try ResponseTyped(fromJSONValue: Json.asObject($0, "ResponseTyped"))
        }
    }

    public var description: String {
        return "result=\(result)"
    }
}

File: Rust/ExampleTests/common/response_typed.rs
#![allow(dead_code, unused_imports, unused_variables)]

use super::json;
use super::response_string::ResponseString;
use super::result_typed::ResultTyped;

#[derive(Debug, Clone, Default, PartialEq)]
pub struct ResponseTyped {
    pub result: ResultTyped,
}

impl ResponseTyped {
    pub fn from_str_struct(s: &ResponseString) -> Self {
        Self {
            result: ResultTyped::from_str_struct(&s.result),
        }
    }

    pub fn to_str_struct(&self) -> ResponseString {
        ResponseString {
            result: self.result.to_str_struct(),
        }
    }

    pub fn to_string_list(list: &[ResponseTyped]) -> Vec<ResponseString> {
        list.iter().map(|t| t.to_str_struct()).collect()
    }

    pub fn from_string_list(list: &[ResponseString]) -> Vec<ResponseTyped> {
        list.iter().map(ResponseTyped::from_str_struct).collect()
    }

    pub fn to_json_value(&self) -> json::Value {
        json::Value::Object(vec![
            ("result".to_string(), self.result.to_json_value()),
        ])
    }

    pub fn to_json(&self) -> String {
        json::write(&self.to_json_value())
    }

    pub fn from_json_value(v: &json::Value) -> json::JsonResult<Self> {
        Ok(Self {
            result: ResultTyped::from_json_value(json::require(v, "result")?)?,
        })
    }

    pub fn from_json(text: &str) -> json::JsonResult<Self> {
        Self::from_json_value(&json::parse(text)?)
    }

    pub fn to_json_list(list: &[ResponseTyped]) -> String {
        json::write(&json::Value::Array(
            list.iter().map(|item| item.to_json_value()).collect(),
        ))
    }

    pub fn from_json_list(text: &str) -> json::JsonResult<Vec<ResponseTyped>> {
        let root = json::parse(text)?;
        let items = json::as_array(&root, "ResponseTyped")?;
        let mut result = Vec::with_capacity(items.len());
        for e in items {
            result.push(Self::from_json_value(e)?);
        }
        Ok(result)
    }
}
